src/screens/ProfileScreen/index.js

```diff
@@ -633,19 +633,19 @@ const ProfileScreen = ({route}) => {
     setIsOptionModalOpen(true);
   };
 
-  const removePostByIdFromContext = () => {
-    const deletedIndex = feeds?.findIndex((find) => selectedPostForOption?.id === find?.id);
+  const removePostByIdFromContext = (post) => {
+    const deletedIndex = feeds?.findIndex((find) => post?.id === find?.id);
     const newData = [...feeds];
     newData?.splice(deletedIndex, 1);
     setMyProfileFeed(newData, myProfileDispatch);
   };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for removing a post from the context seems to be correct. However, it's important to handle the case where `deletedIndex` is -1, which means the post was not found in the feeds. If you try to splice at index -1 without checking, it will remove the last item in the array, which is incorrect behavior.
```diff
+    if (deletedIndex !== -1) {
        const newData = [...feeds];
        newData?.splice(deletedIndex, 1);
        setMyProfileFeed(newData, myProfileDispatch);
+    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->